app/controllers/admin/documents_controller.rb

```diff
@@ -9,4 +9,19 @@ def by_content_id
       redirect_to url_maker.admin_editions_path
     end
   end
+
+  def edit_needs
+    @edition = Edition.find(params[:edition_id])
+    @document = @edition.document
+  end
+
+  def update_needs
+    @edition = Edition.find(params[:edition_id])
+    document = @edition.document
+
+    document.need_ids = params[:need_ids] || []
+    document.patch_meets_user_needs_links
+
+    redirect_to admin_edition_path(@edition)
+  end
```

Comment:
I've moved the actions for editing/updating needs into a needs_controller. Unfortunately I still  need an edition ID in order to return the person to the right edition show page after they've finished choosing their needs. 

TL;DR The transition from "Edit form for needs" to -> "Show page for edition" requires that I know the edition ID.